scripts/build.ts

```diff
@@ -322,25 +327,45 @@ async function buildJS() {
 		input[`plugins/prism-${id}`] = path.join(SRC_DIR, `plugins/${id}/prism-${id}.ts`);
 	}
 
-	let bundle;
+	// Using multiple bundles here mostly for convenience.
+	let esmBundle;
+	let cjsBundle;
 	try {
-		bundle = await rollup({
+		esmBundle = await rollup({
+			input,
+			output: { dir: './dist' },
+			plugins: [
+				rollupTypescript({
+					module: 'esnext',
+					compilerOptions: { declaration: true, declarationDir: './dist/esm' },
+				}),
+			],
+		});
+
+		cjsBundle = await rollup({
 			input,
-			plugins: [rollupTypescript({ module: 'esnext' })],
+			output: { dir: './dist' },
+			plugins: [
+				rollupTypescript({
+					module: 'esnext',
+					compilerOptions: { declaration: true, declarationDir: './dist/cjs' },
```

Comment:
This is how I added friendly types to the output, thankfully it seems to just work.